vizro-core/src/vizro/models/_page.py

```diff
@@ -176,5 +176,7 @@ def build(self) -> _PageBuildType:
         # Build layout with components
         components_container = _build_inner_layout(self.layout, self.components)
         components_container.children.append(dcc.Store(id=f"{ON_PAGE_LOAD_ACTION_PREFIX}_trigger_{self.id}"))
+        components_container.children.append(dcc.Download(id="vizro_download"))
+        components_container.children.append(dcc.Location(id="vizro_url", refresh="callback-nav"))
         components_container.id = "page-components"
         return html.Div([control_panel, components_container])
```

Comment:
Let's simplify the code like this or another way if we decide to place these components within the `components_container`.

```suggestion
        components_container = _build_inner_layout(self.layout, self.components)
        components_container.id = "page-components"
        
        return html.Div([
            dcc.Store(id=f"{ON_PAGE_LOAD_ACTION_PREFIX}_trigger_{self.id}"),
            dcc.Download(id="vizro_download"),
            dcc.Location(id="vizro_url", refresh="callback-nav"),
            control_panel, 
            components_container,
        ])
```